files/validator.rb

```diff
@@ -0,0 +1,42 @@
+def handler
+  req = Nginx::Request.new
+  uri = req.uri # of form .*/small_light[^\/]*/.*
+  v = Nginx::Var.new
+  threshold = v.small_light_maximum_size
+
+  params, uri_redir = parse_uri(uri)
+  if threshold == "" # TODO not sure it's working
+    Nginx.errlogger(Nginx::LOG_NOTICE, "small_light_maximum_size is not defined!") # TODO the magic number "100" is used in the original code, but its meaning is unclear, so we use LOG_NOTICE.
+    Nginx.redirect uri_redir
+    Nginx.return Nginx::OK
+    return
+  end
+  threshold = threshold.to_i
+
+  bad_request = false
+
+  for param in params
+    key, value = param.split("=")
+
+    if ["cw", "dw", "ch", "dh"].index(key)
+      value = value.to_i
+      if value > threshold
+        Nginx::errlogger(Nginx::LOG_WARN, "Invalid resize parameter, " + key + ": " + value.to_s)
+        bad_request = true
```

Comment:
Sure. Fixed in https://github.com/wantedly/nginx-image-server/pull/44/commits/ccd883fd130e435afb76c690d5546a9d0c62ae52.